File: implementations/java/vanilla/src/main/java/org/cwi/waebric/parser/ast/statement/embedding/PreText.java
package org.cwi.waebric.parser.ast.statement.embedding;

import org.cwi.waebric.parser.ast.SyntaxNode;
import org.cwi.waebric.parser.ast.INodeVisitor;

/**
 * "\"" TextChar* "<" -> PreText
 * 
 * @author Jeroen van Schagen
 * @date 02-06-2009
 */
public class PreText extends SyntaxNode {

	private String text;
	
	public PreText(String text) {
		this.text = text;
	}
	
	public String getText() {
		return text;
	}

	public SyntaxNode[] getChildren() {
		return new SyntaxNode[] { /* No children */ };
	}
	
	@Override
	public <T> T accept(INodeVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
}

File: implementations/csharp/vanilla/parser/ast/Embedding/PreText.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Parser.Ast.Embedding
{
    /// <summary>
    /// Node which contains an PreText
    /// \"TextChar*>
    /// </summary>
    public class PreText : ISyntaxNode
    {
        #region Private Members

        private String Text;

        #endregion
        #region Public Methods

        public void SetText(String text)
        {
            Text = text;
        }

        public String GetText()
        {
            return Text;
        }

        public override String ToString()
        {
            return "\"" + Text + "<";
        }

        public ISyntaxNode[] GetSubNodes()
        {
            return new ISyntaxNode[] { };
        }

        public void AcceptVisitor(ISyntaxNodeVisitor visitor)
        {
            visitor.Visit(this);
        }

        #endregion
    }
}
